### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by Katerina-codes**
My work so far refactoring the Game class.

- Renamed methods to remove 'two'.
- Grouped and extracted logic into methods from runGame method.
- Removed duplication.

@MollieS @Maikon @jsuchy

### CODE REVIEW

File: src/main/game/CommandLine/GameRunner.java [hunk 1]
```diff
@@ -10,6 +10,6 @@ public static void main(String[] args) {
         MoveFactory moves = new MoveFactory();
         Rules rules = new Rules(moves.getMoves());
         Game game = new Game(inputOutput, rules);
-        game.runGameTwo();
+        game.runGame();
     }
 }
```
Comment (Maikon) [inline src/main/game/CommandLine/GameRunner.java]: Missing new line

File: src/main/game/Game.java [hunk 2]
```diff
@@ -13,22 +13,33 @@ public Game(Ui inputOutput, Rules rules) {
...
-        inputOutput.promptForGameMode();
-        String gameMode = inputOutput.getGameMode();
-        inputOutput.askForMoveTwo();
-        Moves playerOneMove = inputOutput.getMove();
+        String gameMode = setGameMode();
+        Moves playerOneMove = getPlayerMove();
+        Moves playerTwoMove = getPlayerTwoMove(gameMode);
+        Result winningMove = rules.findWinningPLayer(playerOneMove, playerTwoMove);
+        inputOutput.announceWinner(winningMove);
+    }
```
Comment (Maikon) [inline src/main/game/Game.java]: Better 👍

File: src/main/game/Game.java [hunk 3]
```diff
@@ -13,22 +13,33 @@ public Game(Ui inputOutput, Rules rules) {
...
+        inputOutput.askForMove();
+        return inputOutput.getMove();
+    }
+
+    private String setGameMode() {
+        inputOutput.promptForGameMode();
+        return inputOutput.getGameMode();
     }
-}
+}
```
Comment (Maikon) [inline src/main/game/Game.java:33]: Missing new line

**Review by Maikon — APPROVED** [review]
Good cleaning up 👍

File: src/main/game/CommandLine/CommandlineUi.java [hunk 4]
```diff
@@ -39,12 +40,30 @@ public Moves getMove() {
...
+            }
+
+            if (language instanceof Greek) {
+                String translatedMove = getTranslatedMove(humanMove);
+                convertedMove = convertMove(translatedMove);
+            } else {
+                convertedMove = convertMove(humanMove);
+            }
         }
+        return convertedMove;
```
Comment (Maikon) [inline src/main/game/CommandLine/CommandlineUi.java]: I think this method has low cohesion as it has _multiple reasons to change_ that are _not related to each other_. For example:
1. It knows about game modes (human move vs computer move)
2. It knows about languages (english vs greek)

It's ok having one method that brings the above two ideas together like a wrapper, but the details of 1. and 2. can be in their own methods.

File: src/main/game/CommandLine/CommandlineUi.java [hunk 5]
```diff
@@ -103,6 +122,23 @@ public void askForLanguage() {
         output.println(english.askForLanguage());
     }
 
+    private String getTranslatedMove(String userMove) {
+        return translations.get(userMove);
+    }
+
+    private boolean languageIsEnglish() {
```
Comment (Maikon) [inline src/main/game/CommandLine/CommandlineUi.java]: I like that you extracted this into a method and you're using it below so 👍 Do you still need these checks however if you're passing the selected language from the start of your program?

File: src/main/game/CommandLine/CommandlineUi.java (hunk 4 above)
Comment (felipesere) [reply src/main/game/CommandLine/CommandlineUi.java]: The name of the method also makes me wonder... Why is there a `..Two` tacked at the end? 
Is it for the second move of a game? Why is it different from the first? 
Is it for _game mode 2_? Why does that need to be different from game mode 1?

File: src/main/game/CommandLine/CommandlineUi.java [hunk 6]
```diff
@@ -39,12 +40,30 @@ public Moves getMove() {
         } catch (IOException e) {
             e.printStackTrace();
         }
-        if (userMove.equals("rock") || userMove.equals("paper") || userMove.equals("scissors")) {
-            return convertMove(userMove);
+        return getConvertedMove(userMove);
+    }
+
+    public Moves getMoveTwo(String gameMode) {
```
Comment (felipesere) [inline src/main/game/CommandLine/CommandlineUi.java]: From the name I infer that there is only a limited number of game modes? Maybe an `enum GameMode { ... }` would be more fitting? Paging Dr. @gabbendorf

Comment (felipesere) [reply src/main/game/CommandLine/CommandlineUi.java]: Also, if we were to elevate `gameMode` above a String, maybe we could have the gameMode know how to play differently so you don't need the if-statement here?

File: src/main/game/CommandLine/CommandlineUi.java [hunk 7]
```diff
@@ -103,6 +122,23 @@ public void askForLanguage() {
...
+    private Moves getConvertedMove(String userMove) {
+        if (languageIsEnglish()) {
+            return convertMove(userMove);
+        } else {
+            String translatedMove = getTranslatedMove(userMove);
+            return convertMove(translatedMove);
+        }
+    }
+
     private void translateToEnglish() {
```
Comment (felipesere) [inline src/main/game/CommandLine/CommandlineUi.java]: How come we need this Map if we are passing the language in through the constructor? Is this a hint at that the `Language` class (with English and Greek) does not yet fully model what "language" means our application?

File: src/main/game/Game.java [hunk 8]
```diff
@@ -15,20 +14,23 @@ public Game(Ui inputOutput, Rules rules) {
...
-        String gameMode = inputOutput.getGameMode();
-        inputOutput.askForMoveTwo();
-        Moves playerOneMove = inputOutput.getMove();
+        String gameMode = setGameMode();
+        Moves playerOneMove = getPlayerMove();
+        Moves playerTwoMove = getPlayerTwoMove(gameMode);
+        Result winningMove = rules.findWinningPLayer(playerOneMove, playerTwoMove);
+        inputOutput.announceWinner(winningMove);
+    }
+    private Moves getPlayerTwoMove(String gameMode) {
```
Comment (felipesere) [inline src/main/game/Game.java]: Same `...Two` question as above...

File: src/main/game/CommandLine/CommandlineUi.java [hunk 9]
```diff
@@ -103,6 +122,23 @@ public void askForLanguage() {
...
+
+    private boolean languageIsEnglish() {
+        return language instanceof English;
+    }
+
+    private Moves getConvertedMove(String userMove) {
+        if (languageIsEnglish()) {
+            return convertMove(userMove);
+        } else {
+            String translatedMove = getTranslatedMove(userMove);
```
Comment (felipesere) [inline src/main/game/CommandLine/CommandlineUi.java]: How come we need to "work this hard" to get users move into a proper class `Moves`?
Also, why do we need bespoke methods if `Language` already understands that things might be English or Greek?

**Pull request merged**